File: middleware_loader/core/domain/dtos/request/task_dto.go
package request_dtos

import (
	"middleware_loader/infrastructure/graph/model"

	"github.com/devfeel/mapper"
)

type CreateTaskRequestDTO struct {
	Title        string   `json:"title"`
	Description  string   `json:"description"`
	Priority     []string `json:"priority"`
	Status       string   `json:"status"`
	StartDate    string   `json:"startDate"`
	Deadline     string   `json:"deadline"`
	Duration     string   `json:"duration"`
	ActiveStatus string   `json:"activeStatus"`
	GroupTaskId  string   `json:"groupTaskId"`
}

type UpdateTaskRequestDTO struct {
	UserId         string   `json:"userId"`
	TaskId         string   `json:"taskId"`
	Title          string   `json:"title"`
	Description    string   `json:"description"`
	StartDate      string   `json:"startDate"`
	Deadline       string   `json:"deadline"`
	Duration       string   `json:"duration"`
	Status         string   `json:"status"`
	Priority       []string `json:"priority"`
	TaskOrder      float64  `json:"taskOrder"`
	StopTime       float64   `json:"stopTime"`
	ScheduleTaskId string   `json:"scheduleTaskId"`
}

type GenerateTaskRequestDTO struct {
	Title        string   `json:"title"`
	Description  string   `json:"description"`
	Priority     []string `json:"priority"`
	Status       string   `json:"status"`
	StartDate    string   `json:"startDate"`
	Deadline     string   `json:"deadline"`
	Duration     string   `json:"duration"`
	ActiveStatus string   `json:"activeStatus"`
	ProjectID    string   `json:"projectId"`
}

type UpdateTaskInDialogRequestDTO struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	Status      string `json:"status"`
	TaskID      string `json:"taskId"`
}

type MoveTaskRequestDTO struct {
	OldGroupTaskID string `json:"oldGroupTaskId"`
	NewGroupTaskID string `json:"newGroupTaskId"`
	TaskID         string `json:"taskId"`
}

func NewCreateTaskRequestDTO() *CreateTaskRequestDTO {
	return &CreateTaskRequestDTO{}
}
func (in *CreateTaskRequestDTO) MapperToModel(input model.CreateTaskInput) {
	mapper.AutoMapper(&input, in)
}

func NewUpdateTaskRequestDTO() *UpdateTaskRequestDTO {
	return &UpdateTaskRequestDTO{}
}
func (in *UpdateTaskRequestDTO) MapperToModel(input model.UpdateTaskInput) {
	mapper.AutoMapper(&input, in)
}

func NewGenerateTaskRequestDTO() *GenerateTaskRequestDTO {
	return &GenerateTaskRequestDTO{}
}
func (in *GenerateTaskRequestDTO) MapperToModel(input model.GenerateTaskWithoutGroupTaskInput) {
	mapper.AutoMapper(&input, in)
}

func NewUpdateTaskInDialogInput() *UpdateTaskInDialogRequestDTO {
	return &UpdateTaskInDialogRequestDTO{}
}
func (in *UpdateTaskInDialogRequestDTO) MapperToModel(input model.UpdateTaskInput) {
	mapper.AutoMapper(&input, in)
}

func NewMoveTaskInput() *MoveTaskRequestDTO {
	return &MoveTaskRequestDTO{}
}
func (in *MoveTaskRequestDTO) MapperToModel(input model.MoveTaskInput) {
	mapper.AutoMapper(&input, in)
}

File: person_task_manager/server/src/core/domain/dtos/task.dto.ts
import { IsOptional, IsString } from "class-validator";
import { TaskTagRequest } from "./request_dtos/tag.dto";

export class TaskRequestDto {
    @IsString()
    title!: string;
    @IsString()
    @IsOptional()
    description?: string;
    @IsOptional()
    priority?: string[];
    @IsString()
    @IsOptional()
    status?: string;
    @IsString()
    @IsOptional()
    startDate?: Date;
    @IsString()
    @IsOptional()
    deadline?: Date;
    @IsString()
    @IsOptional()
    duration?: number;
    @IsOptional()
    subTasks?: string[];
    @IsOptional()
    comments?: string[];
    @IsString()
    @IsOptional()
    activeStatus?: string;
    @IsString()
    @IsOptional()
    tag?: string;
}

export class UpdateTaskRequestDto {
    @IsString()
    userId!: string;
    @IsString()
    taskId!: string;
    @IsString()
    title!: string;
    @IsString()
    description!: string;
    @IsString()
    startDate!: string;
    @IsString()
    deadline!: string;
    @IsString()
    duration!: string;
    @IsString()
    status!: string;
    @IsString()
    priority!: string[];
    @IsString()
    taskOrder!: number;
    @IsString()
    stopTime!: number;
    @IsString()
    scheduleTaskId!: string;
    @IsOptional()
    subTasks?: string[];
    @IsOptional()
    comments?: string[];
    @IsString()
    @IsOptional()
    activeStatus?: string;
    @IsString()
    @IsOptional()
    tag?: string;
}

export class GenerateTaskFromScratchRequestDTO {
    @IsString()
    title!: string;
    @IsString()
    @IsOptional()
    description?: string;
    @IsOptional()
    prioriry?: string[];
    @IsString()
    @IsOptional()
    status?: string;
    @IsString()
    @IsOptional()
    startDate?: Date;
    @IsString()
    @IsOptional()
    deadline?: Date;
    @IsString()
    @IsOptional()
    duration?: number;
    @IsString()
    projectId!: string;
    @IsString()
    @IsOptional()
    activeStatus?: string;
    @IsOptional()
    tag?: TaskTagRequest;
}

export class UpdateTaskInDialogDTO {
    @IsString()
    title!: string;
    @IsString()
    @IsOptional()
    description?: string;
    tag!: TaskTagRequest;
    @IsString()
    status!: string;
}

export class TaskDetailRequestDTO {
    userId!: number;
    @IsOptional()
    @IsString()
    taskId?: string;
    @IsOptional()
    @IsString()
    scheduleTaskId?: string;
    @IsString()
    taskDetailType!: string;
}